### PR TITLE
Fixes issue with scoring

### PR DESCRIPTION
**Pull request by paullewis**
The default config set value rather than rawValue for TTI, and while there was a check for type errors of this nature, it came after another early exit, meaning that TTI wasn't included in the report total.

This fixes the issue, and it also makes Lighthouse throw in these situations as they occur at author time when someone updates the default config etc.

### CODE REVIEW

File: lighthouse-core/audits/time-to-interactive.js [hunk 1]
```diff
@@ -121,7 +121,7 @@ class TTIMetric extends Audit {
         // Grab this latency and try the threshold again
         currentLatency = estLatency;
       }
-      const timeToInteractive = startTime.toFixed(1);
+      const timeToInteractive = parseFloat(startTime.toFixed(1));
```
Comment (brendankenny) [inline lighthouse-core/audits/time-to-interactive.js]: this is slightly horrifying (slightly more that it was working. thx JS). While you're in here, can you change it to pass `startTime` itself into `distribution.computeComplementaryPercentile()` instead of the rounded `timeToInteractive`?

File: lighthouse-core/aggregator/aggregate.js [hunk 2]
```diff
@@ -75,16 +75,15 @@ class Aggregate {
     if (typeof expected === 'undefined' ||
         typeof expected.rawValue === 'undefined' ||
         typeof expected.weight === 'undefined') {
-      return weight;
+      throw new Error('Expected is undefined or does not contain rawValue and weight properties');
```
Comment (brendankenny) [inline lighthouse-core/aggregator/aggregate.js]: Is there a way to rephrase this to be clearer to an author of a bad config file? "Expected" isn't going to ring a bell if you see it in a log

File: lighthouse-core/aggregator/aggregate.js [hunk 3]
```diff
@@ -75,16 +75,15 @@ class Aggregate {
...
     }
 
     if (typeof result === 'undefined' ||
-        typeof result.rawValue === 'undefined' ||
         typeof result.score === 'undefined') {
-      return weight;
+      throw new Error('Result is undefined or does not contain score property');
```
Comment (brendankenny) [inline lighthouse-core/aggregator/aggregate.js]: same as above. Should we surface `debugString` at this point if it exists? That way it's at least clear the issue is in the audit or gatherer

File: lighthouse-core/test/aggregator/aggregate.js [hunk 4]
```diff
@@ -125,16 +125,16 @@ describe('Aggregate', () => {
       'Cannot remap: test already exists');
   });
 
-  it('returns a weight of zero for undefined inputs', () => {
-    return assert.equal(Aggregate._convertToWeight(), 0);
+  it('throws for undefined inputs', () => {
+    return assert.throws(_ => Aggregate._convertToWeight(), 0);
```
Comment (brendankenny) [inline lighthouse-core/test/aggregator/aggregate.js]: Maybe also add a test of `Aggregate.compare()` with an expected value missing a `rawValue` property?

**Comment by brendankenny** [comment]
Nice catch. Maybe tweak the exception messages and add another test and looks good.

File: lighthouse-core/audits/time-to-interactive.js (hunk 1 above)
Comment (paullewis) [inline lighthouse-core/audits/time-to-interactive.js]: Done

File: lighthouse-core/aggregator/aggregate.js (hunk 2 above)
Comment (paullewis) [inline lighthouse-core/aggregator/aggregate.js]: Done

File: lighthouse-core/aggregator/aggregate.js (hunk 3 above)
Comment (paullewis) [inline lighthouse-core/aggregator/aggregate.js]: Done

**Comment by paullewis** [comment]
PTAL

File: lighthouse-core/aggregator/aggregate.js [hunk 5]
```diff
@@ -67,24 +67,28 @@ class Aggregate {
...
-  static _convertToWeight(result, expected) {
+  static _convertToWeight(result, expected, name) {
     let weight = 0;
 
     if (typeof expected === 'undefined' ||
         typeof expected.rawValue === 'undefined' ||
         typeof expected.weight === 'undefined') {
-      return weight;
+      const msg =
+          `Config for ${name} is undefined or does not contain rawValue and weight properties`;
```
Comment (brendankenny) [inline lighthouse-core/aggregator/aggregate.js]: maybe add 'aggregations: ' to the beginning of these strings to indicate where in the config this is referring to?

**Comment by brendankenny** [comment]
one last (possible) nit, but LGTM
